### ISSUE TITLE
CodeFile property in dna should fallback to default based on convention

### ISSUE THREAD

**Issue by thedavidmeister** [holochain/holochain-proto#660]
https://developer.holochain.org/HoloWorld_Tutorial_Writing_the_application_DNA

> Change the “CodeFile” property to “readerWriter.js”. This follows a convention of naming your CodeFile the same as your Zome name, which is recommended (but not enforced).

can `CodeFile` have the zome name merged in as a default if not set?

**Comment by Connoropolous** [comment]
That's an idea. I think there's a leaning in the design pattern towards being more explicit than implicit, which is sometimes annoying, but it has advantages too.

**Comment by thedavidmeister** [comment]
@Connoropolous does providing a default make something less explicit?

it still has an explicit value and can be set as needed

when i think of problems caused by implicit things, it's usually more like "emergent behaviour that i can't change and/or reference directly"

**Comment by Connoropolous** [comment]
Ok so if we were to expand this idea...

You can leave CodeFile out of the dna file. 
If you do, it will assume a CodeFile with the name being the same as the zome name, and the file extension being based on the value of `RibosomeType`.
If you do, you MUST name the code file in the zome according to this pattern, otherwise it won't work.

If you do specify it, it will look for the file with the name that you specify.

**Comment by thedavidmeister** [comment]
yeah, although why not use the file extension as is so we don't have to set the type either, disallowing duplicate filenames?

**Comment by Connoropolous** [comment]
not sure what you mean. One way or another we need them to specify the ribosome type, whether that is by checking the file extension in codefile, or by having them set ribosometype.

**Comment by thedavidmeister** [comment]
yeah, that's what i meant
